### ISSUE TITLE
Segmentation fault when running RUST_BACKTRACE=1 on stripped binary

### ISSUE THREAD

**Issue by johnthagen** [rust-lang/rust#44379]
```rust
fn main() {
    panic!();
}
```

```bash
$ cargo run --release
    Finished release [optimized] target(s) in 0.0 secs
     Running `target/release/rust-test`
thread 'main' panicked at 'explicit panic', src/main.rs:2:4
note: Run with `RUST_BACKTRACE=1` for a backtrace.
$ strip target/release/rust-test
$ RUST_BACKTRACE=1 ./target/release/rust-test 
thread 'main' panicked at 'explicit panic', src/main.rs:2:4
Segmentation fault: 11
```

Should this be happening? I'm not surprised `strip`ping would affect the backtrace, but a segmentation fault feels like a bug.

Tested on:
```
active toolchain
----------------

stable-x86_64-apple-darwin (default)
rustc 1.20.0 (f3d6973f4 2017-08-27)
```

**Comment by alexcrichton** [comment]
Thanks for the report and definitely a bug! Should be fixed in https://github.com/rust-lang/rust/issues/44379

**Comment by johnthagen** [comment]
For others tracking, should be fixed in #44384 😄
